timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py

```diff
@@ -399,6 +401,123 @@ def split_by_time(self, cutoff_time: pd.Timestamp) -> Tuple[TimeSeriesDataFrame,
         after._cached_freq = self._cached_freq
         return before, after
 
+    def slice_items_by_index(self, start_index: Union[int, None], end_index: Union[int, None]) -> TimeSeriesDataFrame:
```

Comment:
again 1/ I think the word `index` is overloaded and increases ambiguity over timestep 2/ I don't see why we specify `items` in the name. I don't have a strong opinion on 1 though.

Reply:
So how about we name the new methods `slice_by_index` and `slice_by_time`? I don't think `slice_by_index` is that ambiguous since the function accepts two integers, so it shouldn't be confused with a `pd.Index` object.

I like this option more than keeping the old name `slice_by_timestep` + adding some logic to handle both `slice` and `(int, int)` objects passed to the function.

Reply:
I still think it is confusing. If we iterate over each individual item, the _index_ of those items are time.